### DISCUSSION TITLE
'is None' in hybrid_property causes InvalidRequestError

### DISCUSSION THREAD

**Discussion by davetapley** [sqlalchemy/sqlalchemy#7108] [Usage Questions]
Not sure if this _quite_ qualifies as a bug, so asking here first.

I have:
```python
class Contact(db.Model):
    confirmed = Column(DateTime(timezone=True))

    @hybrid_property
    def not_confirmed(self):
        return self.confirmed == None
```

Which works, but `prospector` doesn't like it, suggesting:

```
comparison to None should be 'if cond is None:'prospector - pep8(E711)
Comparison to None should be 'expr is None'prospector - pylint(singleton-comparison)
```

So I can change it to:

```python
    @hybrid_property
    def not_confirmed(self):
        return self.confirmed is None
```

But then I get:

```
sqlalchemy.exc.InvalidRequestError: When interpreting attribute "Contact.not_confirmed" as a SQL expression, expected __clause_element__() to return a ClauseElement object, got: False
```

---

I can workaround it  with ⬇️, but I couldn't find other reports of this issue, which makes me wonder if I'm doing Something Silly? 🤔 

``` python
        return self.confirmed == None # pylint: disable=singleton-comparison  # nopep8
```

**Comment by CaselIT** [comment]
Hi,

The `==` is the correct one. This works by overriding what the `==` does to generate boolean expression that gets rendered to sql.
Python does not allow overriding `is` so you get a bool right away, that's not what you want when running a sql query

And yes linters complain about it, but there is not much we can do here

**Reply by zzzeek** [reply]
use the is_() method, it should be there and won't generate a warning:  col.is_(None)

https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=is_#sqlalchemy.schema.Column.is_
